financial_calculations/bond_cash_flows_and_pricing.py

```diff
@@ -0,0 +1,105 @@
+import numpy as np
+import pandas as pd
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from utils import (
+    get_ZCB_vector
+)
+
+SPOT_SEMI_BOND_HALF_YEAR = 183
+
+def create_spot_semi_bond_cash_flows(market_close_date, balance, coupon, maturity_years):
```

Comment:
I think we should remove the term 'spot' and just call it create_semi_bond_cash_flows.  the first argument should be changed from market_close_date to effective_date.  this is the start of the first accrual period.  when you need a spot bond, you just call the function with the market_close_date as the effective_date.  since this cash flows are deterministic you don't actually care about the market close date.